OrbitGl/PickingManager.cpp

```diff
@@ -7,28 +7,37 @@
 #include "OpenGl.h"
 #include "OrbitBase/Logging.h"
 
-PickingId PickingManager::CreatePickableId(std::weak_ptr<Pickable> a_Pickable,
-                                           BatcherId batcher_id) {
+PickingId PickingManager::CreateOrGetPickableId(
+    std::weak_ptr<Pickable> pickable, BatcherId batcher_id) {
   absl::MutexLock lock(&mutex_);
-  ++id_counter_;
-  PickingId id =
-      PickingId::Create(PickingType::kPickable, id_counter_, batcher_id);
-  id_pickable_map_[id_counter_] = a_Pickable;
+  uint32_t pickable_id = 0;
+  auto it = pickable_pid_map_.find(pickable.lock().get());
```

Comment:
lock and check here then use shared_ptr.

Reply:
Done